### PR TITLE
Allow ShapeInfo to understand Hollow Cylinder Geometry

### PR DESCRIPTION
**Pull request by aybamidele**
**Description of work.**
I updated the ShapeInfo.h file to account for Hollow Cylinder Geometries. I also included unit tests.

**To test:**

Refs: #25149 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by peterfpeterson** [comment]
Is this intending on adding the explicit calculation to [`Rasterize`](https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Rasterize.cpp) as well?

**Comment by aybamidele** [comment]
@peterfpeterson Yes, it is, but not in this PR.

**Review by martyngigg — CHANGES_REQUESTED** [review]
This looks good and is a good first step to having full annular ring support. :+1: 

One minor thing that I think is worth adding is that in the other `set` functions such as `setCone` etc I think we should extend their unit tests to confirm that the inner and outer radius is set to `0` when they are called.

**Comment by aybamidele** [comment]
Is it okay to do this in the second branch?

**Comment by martyngigg** [comment]
Can we do it here as the work to define the functionality is here?

**Review by martyngigg — DISMISSED** [review]
Changes look good now.

The build failure is the current problem with the RHEL 7 build and doctest failure and is unrelated to these changes.

**Comment by aybamidele** [comment]
Yes, Dimitar just let me know. Thank you.

File: Framework/Geometry/test/ShapeInfoTest.h [hunk 1]
```diff
@@ -122,6 +155,33 @@ class ShapeInfoTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(tshape, ShapeInfo::GeometryShape::SPHERE);
   }
 
+  void testGetObjectGeometryForCylinders() {
+    ShapeInfo shapeInfo;
+    V3D centreBottomBase(0, 0, 0);
```
Comment (DanNixon) [inline Framework/Geometry/test/ShapeInfoTest.h]: Could these test parameters be `constexpr`?

File: Framework/Geometry/test/ShapeInfoTest.h [hunk 2]
```diff
@@ -122,6 +155,33 @@ class ShapeInfoTest : public CxxTest::TestSuite {
...
+    V3D symmetryAxis(0, 1, 1);
+    double innerRadius = 1;
+    double outerRadius = 2;
+    double height = 5;
+    shapeInfo.setHollowCylinder(centreBottomBase, symmetryAxis, innerRadius,
+                                outerRadius, height);
+
+    ShapeInfo::GeometryShape tshape;
+    std::vector<V3D> tpoints;
+    double theight;
```
Comment (DanNixon) [inline Framework/Geometry/test/ShapeInfoTest.h]: Could you use some more verbose variable names here?

e.g. just expanding `t` to `test`.

**Review by DanNixon — CHANGES_REQUESTED** [review]
A couple of small style comments.

**Review by DanNixon — APPROVED** [review]

**Pull request merged**